### PR TITLE
feat(system-addon) Implement section info option. Closes #2875.

### LINKED ISSUE #2875: Implement info option for sections
We would like to have an info option that shows up on hover for Sections, including the ability to have a subtitle, text, and links.

@aaronrbenson is going to supply a design for the hover-over state.

### PR DESCRIPTION
**Pull request by AdamHillier**

### CODE REVIEW

File: system-addon/content-src/components/Sections/Sections.jsx [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+    const {title, initialized, rows, infoOption} = this.props;
     // <Section> <-- React component
     // <section> <-- HTML5 element
     // Dummy component, finished card component needs to be substituted in here
     return (<section>
-        <h3 className="section-title">{title}</h3>
+        <div className="section-top-bar">
+          <h3 className="section-title"><FormattedMessage {...title} /></h3>
+          {infoOption && <span className="section-info-option">
+            <span className="sr-only">Info</span>
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/Sections.jsx]: This needs to be localized

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    .section-title {
+      float: left;
+    }
+
+    .section-info-option {
+      float: right;
+    }
+
+    .info-option-icon {
+      background-image: url('assets/glyph-info-option-12.svg');
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: use the `$image-path` variable here

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      background-position: center;
+      height: 16px;
+      width: 16px;
+      display: inline-block;
+    }
+
+    .section-info-option div {
+      visibility: hidden;
+      opacity: 0;
+      transition: visibility 0.5s, opacity 0.5s ease-in-out;
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: Can you make this 0.2s and add a delay of maybe 0.5s on the transition out only?

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      color: $info-option-text-color;
+      line-height: 120%;
+      width: 320px;
+      right: 0;
+      top: 34px;
+      margin-top: -4px;
+      margin-right: -4px;
+      padding: 24px;
+      user-select: none;
+      -moz-user-select: none;
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: Any reason this prefix needs to be here?

File: system-addon/content-src/components/Sections/_Sections.scss [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    .info-option-header {
+      font-size: 15px;
+      font-weight: 600; /* Semi-bold */
```
Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: IMO I don't think you need this comment

File: system-addon/content-src/components/Sections/_Sections.scss (hunk 4 above)
Comment (AdamHillier) [inline system-addon/content-src/components/Sections/_Sections.scss]: It was in the test pilot version, but I've checked and I don't think Firefox supports it unprefixed

Comment (k88hudson) [inline system-addon/content-src/components/Sections/_Sections.scss]: Still prefixed! 😱 

Alright, leave the un-prefixed for future-proofing then.

File: locales/en-US/strings.properties [hunk 6]
```diff
@@ -65,6 +65,10 @@ search_header={search_engine_name} Search
 search_web_placeholder=Search the Web
 search_settings=Change Search Settings
 
+# LOCALIZATION NOTE (): This is the screenreader only text for the section info
```
Comment (k88hudson) [inline locales/en-US/strings.properties]: `for the section helper text` might be better

**Comment by as-pine-proxy** [comment]
[Treeherder: db9caeb744a09f2f130baa451ea64b03784d6299](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=db9caeb744a09f2f130baa451ea64b03784d6299)

File: locales/en-US/strings.properties [hunk 7]
```diff
@@ -65,6 +65,10 @@ search_header={search_engine_name} Search
 search_web_placeholder=Search the Web
 search_settings=Change Search Settings
 
+# LOCALIZATION NOTE (): This is the screenreader only text for the section
```
Comment (Mardak) [inline locales/en-US/strings.properties]: Typically some string name or prefix goes inside the `()`s

Comment (AdamHillier) [inline locales/en-US/strings.properties]: Ah that's a typo that wasn't caught, I'll sort it out

**Pull request merged**